packages/reactivity/src/effectScope.ts

```diff
@@ -51,40 +51,31 @@ export class EffectScope {
   }
 
   pause(): void {
-    if (this._active) {
-      this._isPaused = true
-      let i, l
-      if (this.scopes) {
-        for (i = 0, l = this.scopes.length; i < l; i++) {
-          this.scopes[i].pause()
-        }
-      }
-      for (i = 0, l = this.effects.length; i < l; i++) {
-        this.effects[i].pause()
-      }
-    }
+    this.forItem('pause')
   }
 
   /**
    * Resumes the effect scope, including all child scopes and effects.
    */
   resume(): void {
-    if (this._active) {
-      if (this._isPaused) {
-        this._isPaused = false
-        let i, l
-        if (this.scopes) {
-          for (i = 0, l = this.scopes.length; i < l; i++) {
-            this.scopes[i].resume()
-          }
-        }
-        for (i = 0, l = this.effects.length; i < l; i++) {
-          this.effects[i].resume()
-        }
-      }
+    if (this._isPaused) {
+      this.forItem('resume')
     }
   }
 
+  forItem(type: 'pause' | 'resume'): void {
```

Comment:
I don't think it's a good change. The logic is not as clear as it used to be. And there's no performance boost, it's not worth it,